wremnants/include/utils.h

```diff
@@ -90,6 +90,14 @@ double deltaR2(float eta1, float phi1, float eta2, float phi2) {
     return deta*deta + dphi*dphi;
 }
 
+RVec<double> vectDeltaR2(RVec<float> eta1, RVec<float> phi1, RVec<float> eta2, RVec<float> phi2) {
+	RVec<double> vect;
+	for (unsigned int i=0U; i!=eta1.size(); i++) {
+    	vect.push_back(deltaR2(eta1[i],phi1[i],eta2[i],phi2[i]));
```

Comment:
It would be more efficient to create an empty array with size equal to eta1 and change each cell in the loop rather than using push_back.
For consistency with other functions you could use Vec_f instead of RVec<float> (they are defined in wremnants/include/defines.h)
You can then start the function with Vec_d vect(eta1.size(), 0); // initialize to 0 or whatever reasonable default value

Reply:
push_back (or better emplace_back) is fine in general, but you should use the "reserve" functionality

e.g. 
```cpp
RVec<double> vect;
vect.reserve(eta1.size())
for (unsigned int i=0U; i!=eta1.size(); i++) {
    	vect.push_back(deltaR2(eta1[i],phi1[i],eta2[i],phi2[i]));
}
```

anyway not critical to fix right this minute.